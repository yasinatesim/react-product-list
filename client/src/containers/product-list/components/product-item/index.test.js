/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import ProductItem from './index';

describe('product item component', () => {
  it('normal case', () => {
    const tree = renderer
      .create(
        <ProductItem
          key={'3782683247'}
          id={'3782683247'}
          name="Test product"
          image="test url"
          price={278.93}
          url="test url"
          cargoType={'1'}
          cargoName="Ucretsiz Kargo"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with integer "id", "key" and "cargoType" values', () => {
    const tree = renderer
      .create(
        <ProductItem
          key={3782683247}
          id={3782683247}
          name="Test product"
          image="test url"
          price={278.93}
          url="test url"
          cargoType={1}
          cargoName="Ucretsiz Kargo"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with liked true', () => {
    const tree = renderer
      .create(
        <ProductItem
          key={3782683247}
          id={3782683247}
          name="Test product"
          image="test url"
          price={278.93}
          url="test url"
          cargoType={1}
          cargoName="Ucretsiz Kargo"
          liked
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
