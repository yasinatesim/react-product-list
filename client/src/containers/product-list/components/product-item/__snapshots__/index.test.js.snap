// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product item component normal case 1`] = `
<div
  className="sc-AxirZ gqAjaP"
>
  <button
    className="sc-AxjAm dugjOI"
    onClick={[Function]}
  >
    <svg
      fill="#ccc"
      height={24}
      width={24}
    >
      <path
        d="M7 3c-1.536 0-3.078.5-4.25 1.7-2.343 2.4-2.279 6.1 0 8.5L12 23l9.25-9.8c2.279-2.4 2.343-6.1 0-8.5-2.343-2.3-6.157-2.3-8.5 0l-.75.8-.75-.8C10.078 3.5 8.535 3 7 3z"
      />
    </svg>
  </button>
  <a
    className="sc-AxiKw cxruOi"
    href="test url"
  >
    <img
      alt="Test product"
      src="test url"
    />
    <div
      className="sc-AxhCb bhsVHd"
    >
      <h3
        className="sc-AxhUy cdCLch"
      >
        <span>
          Test product
        </span>
      </h3>
      <p
        className="sc-AxgMl jstTQD"
      >
        278,93 TL
      </p>
      <div
        className="sc-Axmtr bLvvXp"
      >
        <span
          className="sc-AxheI inDHmN"
        >
          <svg
            fill="#3498db"
            height="22"
            viewBox="0 0 612 612"
            width="22"
          >
            <use
              xlinkHref="#prefix__B"
            />
            <path
              d="M499.918 179.518H410.04c-6.763 0-12.24 5.484-12.24 12.24v238.68c0 6.756 5.477 12.24 12.24 12.24h12.98c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h12.968a12.25 12.25 0 0012.24-12.24v-119.34c0-2.876-1-5.655-2.852-7.852l-99.842-119.34a12.246 12.246 0 00-9.387-4.388zm-77.638 97.92v-61.2c0-6.756 5.477-12.24 12.24-12.24h53.917a12.24 12.24 0 019.4 4.406l50.998 61.2c6.64 7.974.973 20.074-9.406 20.074H434.52a12.24 12.24 0 01-12.24-12.24zM12.24 442.684h31.34c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h153.73c6.757 0 12.24-5.484 12.24-12.24v-306a12.25 12.25 0 00-12.24-12.24H12.24c-6.763 0-12.24 5.484-12.24 12.24v306c0 6.757 5.477 12.24 12.24 12.24z"
            />
            <use
              x={-379.44}
              xlinkHref="#prefix__B"
            />
            <defs>
              <path
                d="M504.9 395.756c-28.684 0-52.02 23.342-52.02 52.02 0 28.684 23.336 52.02 52.02 52.02 28.678 0 52.02-23.336 52.02-52.02 0-28.678-23.342-52.02-52.02-52.02zm0 67.32c-8.44 0-15.3-6.86-15.3-15.3s6.86-15.3 15.3-15.3 15.3 6.86 15.3 15.3-6.86 15.3-15.3 15.3z"
                id="prefix__B"
              />
            </defs>
          </svg>
        </span>
        <span
          className="sc-AxmLO jcpqOg"
        >
          Ucretsiz Kargo
        </span>
      </div>
    </div>
  </a>
</div>
`;

exports[`product item component with integer "id", "key" and "cargoType" values 1`] = `
<div
  className="sc-AxirZ gqAjaP"
>
  <button
    className="sc-AxjAm dugjOI"
    onClick={[Function]}
  >
    <svg
      fill="#ccc"
      height={24}
      width={24}
    >
      <path
        d="M7 3c-1.536 0-3.078.5-4.25 1.7-2.343 2.4-2.279 6.1 0 8.5L12 23l9.25-9.8c2.279-2.4 2.343-6.1 0-8.5-2.343-2.3-6.157-2.3-8.5 0l-.75.8-.75-.8C10.078 3.5 8.535 3 7 3z"
      />
    </svg>
  </button>
  <a
    className="sc-AxiKw cxruOi"
    href="test url"
  >
    <img
      alt="Test product"
      src="test url"
    />
    <div
      className="sc-AxhCb bhsVHd"
    >
      <h3
        className="sc-AxhUy cdCLch"
      >
        <span>
          Test product
        </span>
      </h3>
      <p
        className="sc-AxgMl jstTQD"
      >
        278,93 TL
      </p>
      <div
        className="sc-Axmtr bLvvXp"
      >
        <span
          className="sc-AxheI inDHmN"
        >
          <svg
            fill="#3498db"
            height="22"
            viewBox="0 0 612 612"
            width="22"
          >
            <use
              xlinkHref="#prefix__B"
            />
            <path
              d="M499.918 179.518H410.04c-6.763 0-12.24 5.484-12.24 12.24v238.68c0 6.756 5.477 12.24 12.24 12.24h12.98c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h12.968a12.25 12.25 0 0012.24-12.24v-119.34c0-2.876-1-5.655-2.852-7.852l-99.842-119.34a12.246 12.246 0 00-9.387-4.388zm-77.638 97.92v-61.2c0-6.756 5.477-12.24 12.24-12.24h53.917a12.24 12.24 0 019.4 4.406l50.998 61.2c6.64 7.974.973 20.074-9.406 20.074H434.52a12.24 12.24 0 01-12.24-12.24zM12.24 442.684h31.34c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h153.73c6.757 0 12.24-5.484 12.24-12.24v-306a12.25 12.25 0 00-12.24-12.24H12.24c-6.763 0-12.24 5.484-12.24 12.24v306c0 6.757 5.477 12.24 12.24 12.24z"
            />
            <use
              x={-379.44}
              xlinkHref="#prefix__B"
            />
            <defs>
              <path
                d="M504.9 395.756c-28.684 0-52.02 23.342-52.02 52.02 0 28.684 23.336 52.02 52.02 52.02 28.678 0 52.02-23.336 52.02-52.02 0-28.678-23.342-52.02-52.02-52.02zm0 67.32c-8.44 0-15.3-6.86-15.3-15.3s6.86-15.3 15.3-15.3 15.3 6.86 15.3 15.3-6.86 15.3-15.3 15.3z"
                id="prefix__B"
              />
            </defs>
          </svg>
        </span>
        <span
          className="sc-AxmLO jcpqOg"
        >
          Ucretsiz Kargo
        </span>
      </div>
    </div>
  </a>
</div>
`;

exports[`product item component with integer id and key values 1`] = `
<div
  className="sc-AxirZ gqAjaP"
>
  <button
    className="sc-AxjAm dugjOI"
    onClick={[Function]}
  >
    <svg
      fill="#ccc"
      height={24}
      width={24}
    >
      <path
        d="M7 3c-1.536 0-3.078.5-4.25 1.7-2.343 2.4-2.279 6.1 0 8.5L12 23l9.25-9.8c2.279-2.4 2.343-6.1 0-8.5-2.343-2.3-6.157-2.3-8.5 0l-.75.8-.75-.8C10.078 3.5 8.535 3 7 3z"
      />
    </svg>
  </button>
  <a
    className="sc-AxiKw cxruOi"
    href="test url"
  >
    <img
      alt="Test product"
      src="test url"
    />
    <div
      className="sc-AxhCb bhsVHd"
    >
      <h3
        className="sc-AxhUy cdCLch"
      >
        <span>
          Test product
        </span>
      </h3>
      <p
        className="sc-AxgMl jstTQD"
      >
        278,93 TL
      </p>
      <div
        className="sc-Axmtr bLvvXp"
      >
        <span
          className="sc-AxheI inDHmN"
        >
          <svg
            fill="#3498db"
            height="22"
            viewBox="0 0 612 612"
            width="22"
          >
            <use
              xlinkHref="#prefix__B"
            />
            <path
              d="M499.918 179.518H410.04c-6.763 0-12.24 5.484-12.24 12.24v238.68c0 6.756 5.477 12.24 12.24 12.24h12.98c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h12.968a12.25 12.25 0 0012.24-12.24v-119.34c0-2.876-1-5.655-2.852-7.852l-99.842-119.34a12.246 12.246 0 00-9.387-4.388zm-77.638 97.92v-61.2c0-6.756 5.477-12.24 12.24-12.24h53.917a12.24 12.24 0 019.4 4.406l50.998 61.2c6.64 7.974.973 20.074-9.406 20.074H434.52a12.24 12.24 0 01-12.24-12.24zM12.24 442.684h31.34c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h153.73c6.757 0 12.24-5.484 12.24-12.24v-306a12.25 12.25 0 00-12.24-12.24H12.24c-6.763 0-12.24 5.484-12.24 12.24v306c0 6.757 5.477 12.24 12.24 12.24z"
            />
            <use
              x={-379.44}
              xlinkHref="#prefix__B"
            />
            <defs>
              <path
                d="M504.9 395.756c-28.684 0-52.02 23.342-52.02 52.02 0 28.684 23.336 52.02 52.02 52.02 28.678 0 52.02-23.336 52.02-52.02 0-28.678-23.342-52.02-52.02-52.02zm0 67.32c-8.44 0-15.3-6.86-15.3-15.3s6.86-15.3 15.3-15.3 15.3 6.86 15.3 15.3-6.86 15.3-15.3 15.3z"
                id="prefix__B"
              />
            </defs>
          </svg>
        </span>
        <span
          className="sc-AxmLO jcpqOg"
        >
          Ucretsiz Kargo
        </span>
      </div>
    </div>
  </a>
</div>
`;

exports[`product item component with liked true 1`] = `
<div
  className="sc-AxirZ gqAjaP"
>
  <button
    className="sc-AxjAm fqVtXk"
    onClick={[Function]}
  >
    <svg
      fill="#ccc"
      height={24}
      width={24}
    >
      <path
        d="M7 3c-1.536 0-3.078.5-4.25 1.7-2.343 2.4-2.279 6.1 0 8.5L12 23l9.25-9.8c2.279-2.4 2.343-6.1 0-8.5-2.343-2.3-6.157-2.3-8.5 0l-.75.8-.75-.8C10.078 3.5 8.535 3 7 3z"
      />
    </svg>
  </button>
  <a
    className="sc-AxiKw cxruOi"
    href="test url"
  >
    <img
      alt="Test product"
      src="test url"
    />
    <div
      className="sc-AxhCb bhsVHd"
    >
      <h3
        className="sc-AxhUy cdCLch"
      >
        <span>
          Test product
        </span>
      </h3>
      <p
        className="sc-AxgMl jstTQD"
      >
        278,93 TL
      </p>
      <div
        className="sc-Axmtr bLvvXp"
      >
        <span
          className="sc-AxheI inDHmN"
        >
          <svg
            fill="#3498db"
            height="22"
            viewBox="0 0 612 612"
            width="22"
          >
            <use
              xlinkHref="#prefix__B"
            />
            <path
              d="M499.918 179.518H410.04c-6.763 0-12.24 5.484-12.24 12.24v238.68c0 6.756 5.477 12.24 12.24 12.24h12.98c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h12.968a12.25 12.25 0 0012.24-12.24v-119.34c0-2.876-1-5.655-2.852-7.852l-99.842-119.34a12.246 12.246 0 00-9.387-4.388zm-77.638 97.92v-61.2c0-6.756 5.477-12.24 12.24-12.24h53.917a12.24 12.24 0 019.4 4.406l50.998 61.2c6.64 7.974.973 20.074-9.406 20.074H434.52a12.24 12.24 0 01-12.24-12.24zM12.24 442.684h31.34c6.06 0 11.426-4.364 12.2-10.373 4.804-36.806 34.162-59.633 69.676-59.633s64.872 22.828 69.676 59.633c.783 6 6.144 10.373 12.2 10.373h153.73c6.757 0 12.24-5.484 12.24-12.24v-306a12.25 12.25 0 00-12.24-12.24H12.24c-6.763 0-12.24 5.484-12.24 12.24v306c0 6.757 5.477 12.24 12.24 12.24z"
            />
            <use
              x={-379.44}
              xlinkHref="#prefix__B"
            />
            <defs>
              <path
                d="M504.9 395.756c-28.684 0-52.02 23.342-52.02 52.02 0 28.684 23.336 52.02 52.02 52.02 28.678 0 52.02-23.336 52.02-52.02 0-28.678-23.342-52.02-52.02-52.02zm0 67.32c-8.44 0-15.3-6.86-15.3-15.3s6.86-15.3 15.3-15.3 15.3 6.86 15.3 15.3-6.86 15.3-15.3 15.3z"
                id="prefix__B"
              />
            </defs>
          </svg>
        </span>
        <span
          className="sc-AxmLO jcpqOg"
        >
          Ucretsiz Kargo
        </span>
      </div>
    </div>
  </a>
</div>
`;
